bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/scheduling/Strand.java

```diff
@@ -438,8 +438,14 @@ public String dumpState() {
         }
 
         strandInfo.append(" [");
-        strandInfo.append(this.metadata.getModuleOrg()).append(".").append(this.metadata.getModuleName()).append(".")
-                .append(this.metadata.getModuleVersion()).append(":").append(this.metadata.getParentFunctionName());
+        StrandMetadata strandMetadata = this.metadata;
+        if (strandMetadata == null) {
+            strandInfo.append("N/A");
+        } else {
+            strandInfo.append(strandMetadata.getModuleOrg()).append(".").append(strandMetadata.getModuleName())
```

Comment:
Shall we create an separate issue to handle null values with metadata module org, name etc.

Reply:
Will do.